jetstream/stream_config.go

```diff
@@ -35,30 +35,35 @@ type (
-		MaxMsgs              int64           `json:"max_msgs"`
-		MaxBytes             int64           `json:"max_bytes"`
-		Discard              DiscardPolicy   `json:"discard"`
-		DiscardNewPerSubject bool            `json:"discard_new_per_subject,omitempty"`
-		MaxAge               time.Duration   `json:"max_age"`
-		MaxMsgsPerSubject    int64           `json:"max_msgs_per_subject"`
-		MaxMsgSize           int32           `json:"max_msg_size,omitempty"`
-		Storage              StorageType     `json:"storage"`
-		Replicas             int             `json:"num_replicas"`
-		NoAck                bool            `json:"no_ack,omitempty"`
-		Template             string          `json:"template_owner,omitempty"`
-		Duplicates           time.Duration   `json:"duplicate_window,omitempty"`
-		Placement            *Placement      `json:"placement,omitempty"`
-		Mirror               *StreamSource   `json:"mirror,omitempty"`
-		Sources              []*StreamSource `json:"sources,omitempty"`
-		Sealed               bool            `json:"sealed,omitempty"`
-		DenyDelete           bool            `json:"deny_delete,omitempty"`
-		DenyPurge            bool            `json:"deny_purge,omitempty"`
-		AllowRollup          bool            `json:"allow_rollup_hdrs,omitempty"`
+		Name                 string           `json:"name"`
+		Description          string           `json:"description,omitempty"`
+		Subjects             []string         `json:"subjects,omitempty"`
+		Retention            RetentionPolicy  `json:"retention"`
+		MaxConsumers         int              `json:"max_consumers"`
+		MaxMsgs              int64            `json:"max_msgs"`
+		MaxBytes             int64            `json:"max_bytes"`
+		Discard              DiscardPolicy    `json:"discard"`
+		DiscardNewPerSubject bool             `json:"discard_new_per_subject,omitempty"`
+		MaxAge               time.Duration    `json:"max_age"`
+		MaxMsgsPerSubject    int64            `json:"max_msgs_per_subject"`
+		MaxMsgSize           int32            `json:"max_msg_size,omitempty"`
+		Storage              StorageType      `json:"storage"`
+		Replicas             int              `json:"num_replicas"`
+		NoAck                bool             `json:"no_ack,omitempty"`
+		Template             string           `json:"template_owner,omitempty"`
+		Duplicates           time.Duration    `json:"duplicate_window,omitempty"`
+		Placement            *Placement       `json:"placement,omitempty"`
+		Mirror               *StreamSource    `json:"mirror,omitempty"`
+		Sources              []*StreamSource  `json:"sources,omitempty"`
+		Sealed               bool             `json:"sealed,omitempty"`
+		DenyDelete           bool             `json:"deny_delete,omitempty"`
+		DenyPurge            bool             `json:"deny_purge,omitempty"`
+		AllowRollup          bool             `json:"allow_rollup_hdrs,omitempty"`
+		Compression          StoreCompression `json:"compression"`
```

Comment:
@piotrpio just curious why did you model this as an enum and not just a string, since there's already a custom (un)marshaler that could provide the default values?

Reply:
NM, seems to be the pattern